### ISSUE TITLE
False positive in `e18e/prefer-array-fill`

### ISSUE THREAD

**Issue by sheremet-va** [e18e/eslint-plugin#131]
`e18e/prefer-array-fill` assumes that a function call always returns the same value. This is not always true, I feel like pure functions are actually pretty rare in JS world. For example, `Math.random`:

```ts
const data = Array.from({ length: 10000 }, () => Math.random())
```

Is turned into with an auto fix

```ts
const data = Array.from({ length: 10000 }).fill(Math.random())
```

**Comment by 43081j** [comment]
i feel like we fixed this in other rules too by just skipping any usages containing call expressions.

happy to do the same for this one

**Comment by 43081j** [comment]
@sheremet-va it seems this has never been the case 🤔 

https://github.com/e18e/eslint-plugin/blob/c675b93e2acf98366aaf9175d8ccf9edefa0927b/src/rules/prefer-array-fill.ts#L14-L18

this catches it. if i copy paste your code into a test, it passes fine (doesn't get flagged)
